homeassistant/components/garage_door/zwave.py

```diff
@@ -25,7 +25,7 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
     node = zwave.NETWORK.nodes[discovery_info[zwave.ATTR_NODE_ID]]
     value = node.values[discovery_info[zwave.ATTR_VALUE_ID]]
 
-    if value.command_class != zwave.COMMAND_CLASS_SWITCH_BINARY or \
+    if value.command_class != zwave.COMMAND_CLASS_SWITCH_BINARY and \
```

Comment:
In this case I believe not because the logic is inverted. 

With or the statement short circuits on the first comparison, so it never even checks barrier as the Linear unit only reports as a Barrier, not a switch. This was the problem that took me several hours of debugging to resolve because everything looked right but it would never add the device. Once changed to and it works perfectly.